classes/board.php

```diff
@@ -1264,10 +1264,17 @@ public static function board_is_editor($boardid) {
      * a particular board.
      *
      * @param int $boardid
-     * @return void
+     * @return boolean
      */
     public static function board_users_can_edit($boardid) {
         global $DB;
+
+        $context = static::context_for_board($boardid);
+        if (!has_capability('mod/board:post', $context)) {
+            // the user is not allowed to post via capabilities.
```

Comment:
Sorry for this request, but CI checks are god, and they prefer comments starting with capital letters. TIA!